algo/src/test/java/com/hartwig/serve/sources/ckb/CkbKnownEventsExtractorTest.java

```diff
@@ -0,0 +1,131 @@
+                        .build())
+                .build();
+    }
+
+    @NotNull
+    private static ExtractedEvent createExon(@NotNull String gene, @NotNull String event, @Nullable String proteinEffect) {
+        return ImmutableExtractedEvent.builder()
+                .gene(gene)
+                .event(event)
+                .variant(CkbTestFactory.createVariant(gene, event, gene + " " + event, proteinEffect))
+                .eventType(EventType.EXON)
+                .eventExtractorOutput(ImmutableEventExtractorOutput.builder()
+                        .addExons(ImmutableExonAnnotation.builder()
+                                .from(RangeTestFactory.createRangeAnnotation(gene, "1", 1, 1, MutationType.INFRAME_DELETION))
+                                .inputTranscript("transcript")
+                                .inputExonRank(1)
+                                .build())
+                        .build())
+                .build();
+    }
+
+    @NotNull
+    private static ExtractedEvent createFusion(@NotNull String geneUp, @NotNull String geneDown, @Nullable String proteinEffect) {
+        return ImmutableExtractedEvent.builder()
+                .gene(geneUp)
+                .event(geneDown + "-" + geneUp)
+                .variant(CkbTestFactory.createVariant(geneUp, geneDown + "-" + geneUp, geneDown + "-" + geneUp + " fusion", proteinEffect))
+                .eventType(EventType.FUSION_PAIR)
+                .eventExtractorOutput(ImmutableEventExtractorOutput.builder()
+                        .fusionPair(FusionTestFactory.createFusionPair(geneUp, geneDown, 1, 1))
+                        .build())
+                .build();
+    }
+
+    @NotNull
+    private static ExtractedEvent createHotspot(@NotNull String gene, @NotNull String mutation, @Nullable String proteinEffect) {
+        return ImmutableExtractedEvent.builder()
+                .gene(gene)
+                .event(gene + " " + mutation)
+                .variant(CkbTestFactory.createVariant(gene, mutation, gene + " " + mutation, proteinEffect))
+                .eventType(EventType.VARIANT)
+                .eventExtractorOutput(ImmutableEventExtractorOutput.builder()
+                        .addVariants(HotspotTestFactory.createVariantAnnotation(gene, "1", 1, "T", "A"))
+                        .build())
+                .build();
+    }
+
+    @NotNull
+    private static ExtractedEvent createAmplification(@NotNull String gene, @Nullable String proteinEffect) {
+        return ImmutableExtractedEvent.builder()
+                .gene(gene)
+                .event(gene + " amp")
+                .variant(CkbTestFactory.createVariant(gene, "amp", gene + " amp", proteinEffect))
+                .eventType(EventType.AMPLIFICATION)
+                .eventExtractorOutput(ImmutableEventExtractorOutput.builder()
+                        .copyNumber(ImmutableGeneAnnotationImpl.builder().gene(gene).event(GeneEvent.AMPLIFICATION).build())
```

Comment:
I would use `GeneTestFactory.createGeneAnnotation` here. Even though the impact is minor it is good to minimise the creation of immutables to the least amount of places, that makes it relatively easier to add a field.